### PR TITLE
[Mellanox] Use Debian reboot in Nvidia platform reboot when it is invoked from kdump capture boot

### PR DESCRIPTION
**Pull request by stephenxs**
#### Why I did it

When a kernel crash occurs, the system will reboot to the kdump capture kernel if kdump is enabled (`config kdump enable`). In the kdump capture boot, it only stores the crash information, and then reboot the system to a normal boot.
In this boot, no SONiC service is started but it invokes `reboot` which is actually the SONiC reboot that depends on SONiC services. There is a logic to skip all SONiC stuff and invoke platform reboot in SONiC reboot to avoid issues.
However, on Nvidia platforms, the platform reboot still depends on SONiC services, which can cause issues.
So, the Debian reboot is called directly in platform reboot if it is invoked from the kdump capture boot.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

Manual test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [x] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@lguohan could you please review or suggest someone to review?

**Review by vadimp-nvidia — APPROVED** [review]
Looks fine.

**Review by keboliu — APPROVED** [review]

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot [hunk 1]
```diff
@@ -28,6 +28,17 @@ function SafePwrCycle() {
 
 ParseArguments "$@"
 
+# Reboot immediately if the kdump capture kernel is running
+VMCORE_FILE=/proc/vmcore
+if [ -e $VMCORE_FILE -a -s $VMCORE_FILE ]; then
```
Comment (qiluo-msft) [inline device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot:33]: `-s` include `-e`, so you only need `-s`.

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot [hunk 2]
```diff
@@ -28,6 +28,17 @@ function SafePwrCycle() {
...
+    sync ; sync
```
Comment (qiluo-msft) [inline device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot:34]: Remove blank before `;`

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot [hunk 3]
```diff
@@ -28,6 +28,17 @@ function SafePwrCycle() {
...
+    umount -fa > /dev/null 2&>1
```
Comment (qiluo-msft) [inline device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot:35]: I believe this is `2>&1`. Do you have test output?

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot (hunk 1 above)
Comment (stephenxs) [reply device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot]: fixed

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot (hunk 2 above)
Comment (stephenxs) [reply device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot]: fixed.

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot (hunk 3 above)
Comment (stephenxs) [reply device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot]: fixed. test output [test.log](https://github.com/sonic-net/sonic-buildimage/files/12179397/test.log)

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot (hunk 1 above)
Comment (qiluo-msft) [reply device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot]: What is purpose of first `$VMCORE_FILE`? single `-s` is good enough?

Comment (stephenxs) [reply device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot]: yes. fixed.

**Comment by keboliu** [comment]
@qiluo-msft  all comments are fixed,  would you please take a look?

**Comment by stephenxs** [comment]
Failed due to `bgp/test_bgp_update_timer.py|||2 failed, stop whole test plan` but I do not have the privilege to check the details. In any case, it isn't relevant to the PR. @qiluo-msft  Could you please help to check the failure

**Review by qiluo-msft — APPROVED** [review]

**Comment by StormLiangMS** [comment]
ADO: 24766309

**Comment by stephenxs** [comment]
> @stephenxs cherry pick PR didn't pass PR checker. Please check!!!#16050

Thanks @mssonicbld  but it doesn't look like an issue of the PR itself
In the PR, only Mellanox platform specific reboot util is updated, which won't be reached in the following failing test.
The same failure also fails other PRs being cherry-picked to 202211 like https://github.com/sonic-net/sonic-buildimage/pull/16071
Could you please check whether there is any issue in 202211 azure pipeline?
```
telemetry/test_events.py::test_events[vlab-01] 
-------------------------------- live log call ---------------------------------
05:40:04 __init__.pytest_runtest_call             L0040 ERROR  | Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/_pytest/python.py", line 1761, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/usr/local/lib/python3.8/dist-packages/pluggy/_hooks.py", line 433, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/usr/local/lib/python3.8/dist-packages/pluggy/_manager.py", line 112, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 116, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 80, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python3.8/dist-packages/_pytest/python.py", line 192, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/var/src/sonic-mgmt/tests/telemetry/test_events.py", line 55, in test_events
    module.test_event(duthost, gnxi_path, ptfhost, DATA_DIR, validate_yang)
  File "/var/src/sonic-mgmt/tests/./telemetry/events/swss_events.py", line 25, in test_event
    run_test(duthost, gnxi_path, ptfhost, data_dir, validate_yang, shutdown_interface,
  File "/var/src/sonic-mgmt/tests/./telemetry/events/run_events_test.py", line 29, in run_test
    validate_yang(duthost, dest, tag)
  File "/var/src/sonic-mgmt/tests/telemetry/test_events.py", line 26, in validate_yang
    ret = duthost.shell(cmd)
  File "/var/src/sonic-mgmt/tests/common/devices/multi_asic.py", line 118, in _run_on_asics
    return getattr(self.sonichost, self.multi_asic_attr)(*module_args, **complex_args)
  File "/var/src/sonic-mgmt/tests/common/devices/base.py", line 127, in _run
    raise RunAnsibleModuleFail("run module {} failed".format(self.module_name), res)
tests.common.errors.RunAnsibleModuleFail: run module shell failed, Ansible Results =>
{"changed": true, "cmd": "python ~/validate_yang_events.py -f ~/if_state.json -y sonic-events-swss", "delta": "0:00:00.129535", "end": "2023-08-10 05:40:03.600358", "failed": true, "msg": "non-zero return code", "rc": 1, "start": "2023-08-10 05:40:03.470823", "stderr": "libyang[0]: Leafref \"/sonic-port:sonic-port/sonic-port:PORT/sonic-port:PORT_LIST/sonic-port:name\" of value \"Ethernet24\" points to a non-existing leaf. (path: /sonic-events-swss:sonic-events-swss/if-state/ifname)\nTraceback (most recent call last):\n  File \"/root/validate_yang_events.py\", line 81, in <module>\n    main()\n  File \"/root/validate_yang_events.py\", line 77, in main\n    validator.validateJSON()\n  File \"/root/validate_yang_events.py\", line 64, in validateJSON\n    raise e\n  File \"/root/validate_yang_events.py\", line 60, in validateJSON\n    self.ctx.parse_data_mem(data_json, ly.LYD_JSON,\n  File \"/usr/lib/python3/dist-packages/yang.py\", line 2604, in parse_data_mem\n    return _yang.Context_parse_data_mem(self, data, format, options)\nRuntimeError: Leafref \"/sonic-port:sonic-port/sonic-port:PORT/sonic-port:PORT_LIST/sonic-port:name\" of value \"Ethernet24\" points to a non-existing leaf.", "stderr_lines": ["libyang[0]: Leafref \"/sonic-port:sonic-port/sonic-port:PORT/sonic-port:PORT_LIST/sonic-port:name\" of value \"Ethernet24\" points to a non-existing leaf. (path: /sonic-events-swss:sonic-events-swss/if-state/ifname)", "Traceback (most recent call last):", "  File \"/root/validate_yang_events.py\", line 81, in <module>", "    main()", "  File \"/root/validate_yang_events.py\", line 77, in main", "    validator.validateJSON()", "  File \"/root/validate_yang_events.py\", line 64, in validateJSON", "    raise e", "  File \"/root/validate_yang_events.py\", line 60, in validateJSON", "    self.ctx.parse_data_mem(data_json, ly.LYD_JSON,", "  File \"/usr/lib/python3/dist-packages/yang.py\", line 2604, in parse_data_mem", "    return _yang.Context_parse_data_mem(self, data, format, options)", "RuntimeError: Leafref \"/sonic-port:sonic-port/sonic-port:PORT/sonic-port:PORT_LIST/sonic-port:name\" of value \"Ethernet24\" points to a non-existing leaf."], "stdout": "2023-08-10 05:40:03,588 [INFO] Exception thrown: Leafref \"/sonic-port:sonic-port/sonic-port:PORT/sonic-port:PORT_LIST/sonic-port:name\" of value \"Ethernet24\" points to a non-existing leaf.", "stdout_lines": ["2023-08-10 05:40:03,588 [INFO] Exception thrown: Leafref \"/sonic-port:sonic-port/sonic-port:PORT/sonic-port:PORT_LIST/sonic-port:name\" of value \"Ethernet24\" points to a non-existing leaf."]}

FAILED                                                                   [100%]
```

**Pull request merged**